src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/StateMachineTypeSymbol.cs

```diff
@@ -57,5 +59,37 @@ IMethodSymbol ISynthesizedMethodBodyImplementationSymbol.Method
         {
             get { return KickoffMethod; }
         }
+
+        public sealed override ImmutableArray<CSharpAttributeData> GetAttributes()
+        {
+            if (_attributes.IsDefault)
+            {
+                Debug.Assert(base.GetAttributes().Length == 0);
+
+                ArrayBuilder<CSharpAttributeData> builder = null;
+
+                // Inherit some attributes from the container of the kickoff method
+                var kickoffType = KickoffMethod.ContainingType;
+                foreach (var attribute in kickoffType.GetAttributes())
+                {
+                    if (attribute.IsTargetAttribute(kickoffType, AttributeDescription.DebuggerNonUserCodeAttribute) ||
+                        attribute.IsTargetAttribute(kickoffType, AttributeDescription.DebuggerStepThroughAttribute))
+                    {
+                        if (builder == null)
+                        {
+                            builder = ArrayBuilder<CSharpAttributeData>.GetInstance(4); // only 4 different attributes are inherited at the moment
```

Comment:
The number here is 4 but the conditional above only uses 2 attributes.  Am I missing something?